Lib/unittest/async_case.py

```diff
@@ -79,17 +79,26 @@ async def enterAsyncContext(self, cm):
         return result
 
     def _callSetUp(self):
-        self._asyncioTestContext.run(self.setUp)
-        self._callAsync(self.asyncSetUp)
+        self._setupAsyncioRunner()
+        try:
+            self._asyncioTestContext.run(self.setUp)
+            self._callAsync(self.asyncSetUp)
+        except BaseException:
```

Comment:
Hm, I don't like catching `BaseException` here. It seems to only coincidentally fix the underlying issue -- which IMO is that the asyncio runner is set up even for skipped tests.

And the duplication of calling `_tearDownAsyncioRunner` both here and in `_callTearDown` also jars.

Reply:
Ah yes this means if you have an IsolatedAsyncioTestCase with only skipped tests it means `set_event_loop(None)` is called when you might not have expected it to